Contributions/Using_Linux_in_a_Safe_System.md

```diff
@@ -412,6 +412,64 @@ and/or costly.
     of the SoC watchdog toward multiple operating systems, however it
     could go beyond that, and reach higher levels of complexity.
 
+  * **System Decomposition**
+ 
+    The previous points can be leveraged and extended to cases where the
+    safety requirements rely on multiple subsystems, possibly across different
+    software layers.
+    
+    As usual, it is extremely important to assess upfront the requirements
+    for availability.
+
+    A requirement for simply detecting the potential interference can be easier
+    to deal with, than a requirement to ensure continued safety operations,
+    in the presence of interference.
+
+    Leaving aside, for now, considerations about the latter scenario, the former
+    one still carries a need for the detection to happen within a maximum interval,
+    measured from the beginning of the interference.
+
+    With these two requirements in mind:
+    * detecting interference
+    * detecting the interference within a maximum allowed time
+      
+    One can attempt to isolate those subsystems directly involved, and investigate
+    the possibility of periodically validating its continued integrity.
+    
+    Here, the basic concept is that one could be able to provide - periodically -
+    a preset stimuli, and compare the live output of the system with the
+    pre-calculated expected one, that the system should produce, when free from
+    interference.
+    
+    Taking as example an image recognition pipeline, tasked with receiving an input
+    from a sensor and recognising items from a given set (e.g. a speed limit sign),
+    this could be implemented in the following way:
+    * periodically inject a known image, of a specific object(s)
+    * expect that within a given time the system will correctly identify the object(s)
+    
+    Such an approach has the advantage of not requiring any special partitioning of
+    the architecture (e.g. which parts must be implemented as user processes and
+    which parts must be kernel components).
+    
+    However, to work, it must also satisfy several conditions:
```

Comment:
IMO there is a critical aspect that we are missing here:
For ASIL decomposition to be valid we need to have:
a) Redundant requirements allocated to both the monitored subsystems and the monitors
b) A safety analysis on the monitored subsystems shall verify the completeness and sufficiency of the safety requirements allocated to the monitor and shall also verify the adequacy of the monitor design
c) The monitored subsystems and the monitors must be independent

In the text below you are basically elaborating on point c) while you're missing to elaborate on point a) and b). The main issues being that the higher the complexity or the decomposed requirements and of the monitored code, the higher the complexity and cost of the monitor is. This is an important consideration to be made for choosing or not choosing to afford the redundant monitor design, verification and deployment costs...

Reply:
Costs, intended as monetary costs, are subjective, and likely to change over time, I'd rather leave them out entirely.
Besides, costs can be optimised later on, when the goal is somehow achieved.

Reply:
Perhaps we should have a separate section or even document about costs, if it is an aspect that is considered within the scope of ELISA. I've seen it surfacing in many cases. I'm not oblivious to it, but markets can be so very different that talking about costs can became a very tricky subject. @reiterative @paolonig I gladly leave that to the both of you ;-)